contracts/manager/BaseManager.sol

```diff
@@ -180,9 +317,141 @@ contract BaseManager {
      * @param _module           Module to remove
      */
     function removeModule(address _module) external onlyOperator {
+        require(!protectedModules[_module].isProtected, "Module protected");
         setToken.removeModule(_module);
     }
 
+    /**
+     * OPERATOR ONLY: Marks a currently protected module as unprotected and deletes its authorized
+     * adapter registries. Removes module from the SetToken. Increments the `emergencies` counter,
+     * prohibiting any operator-only module or extension additions until `emergencyReplaceProtectedModule`
+     * is executed.
+     *
+     * Called by operator when a module must be removed immediately for security reasons and it's unsafe
+     * to wait for a `mutualUpgrade` process to play out.
+     *
+     * @param _module           Module to remove
+     */
+    function emergencyRemoveProtectedModule(address _module) external onlyOperator {
+        _unProtectModule(_module);
+        setToken.removeModule(_module);
+        emergencies += 1;
+
+        EmergencyRemovedProtectedModule(_module);
+    }
+
+    /**
+     * OPERATOR ONLY: Marks an existing module as protected and authorizes existing adapters for
+     * it. Adds module to the protected modules list
+     *
+     * The operator uses this when they're adding new features and want to assure the methodologist
+     * they won't be unilaterally changed in the future. Cannot be called during an emergency because
+     * methodologist needs to explicitly approve protection arrangements under those conditions.
+     *
+     * @param  _module          Module to protect
+     * @param  _adapters        Array of adapters to authorize for protected module
+     */
+    function protectModule(address _module, address[] memory _adapters)
```

Comment:
Seems like there's an inconsistency in extension authorization between this and `authorizeAdapter`. Here there's no check to see if the extension has already been added (not authorized!) whereas the `authorizeAdapter` there's an explicit check to see if it has been added:

See Line 259

Reply:
I would tend towards not requiring the extension to have been "added" before being "authorized"

Reply:
As noted above - `authorizeAdapter` and `protectModule` now both automatically add the adapter if missing with ebd8676